dashboard/app/controllers/rubrics_controller.rb

```diff
@@ -97,7 +103,29 @@ def run_ai_evaluations_for_user
     is_level_ai_enabled = EvaluateRubricJob.ai_enabled?(script_level)
     return head :bad_request unless is_level_ai_enabled
 
-    EvaluateRubricJob.perform_later(user_id: user.id, script_level_id: script_level.id)
+    # Create a record of this work request
+    rubric_ai_evaluation = RubricAiEvaluation.create!(
+      user_id: user_id,
+      requester_id: current_user.id,
+      status: SharedConstants::RUBRIC_AI_EVALUATION_STATUS[:QUEUED],
+      project_id: 0,
+    )
+
+    # Find the rubric (must have something to evaluate)
+    rubric = Rubric.find_by!(lesson_id: script_level.lesson.id, level_id: script_level.level.id)
+    return head :bad_request unless rubric
+
+    # Create a blank record of all the pending learning goal ai evaluations
+    rubric.learning_goals.each do |learning_goal|
+      LearningGoalAiEvaluation.create!(
+        rubric_ai_evaluation_id: rubric_ai_evaluation.id,
+        learning_goal_id: learning_goal.id,
+        understanding: RUBRIC_UNDERSTANDING_LEVELS[:UNKNOWN],
+        ai_confidence: 0,
+      )
+    end
```

Comment:
> quoting from the PR description just to help connect the dots:
> 
> > This puts into place creating the RubricAiEvaluation records. When a job is scheduled, it creates one of these base records, sets its status to `QUEUED` and then also creates a set of LearningGoalAiEvaluation records with scores of `-1` to reflect they are not actually assessed.
> 
> if we do this, I'm concerned that the UI will show invalid data (or be completely broken by the new "unknown" value) while the request is processing, rather than the safer option of showing the old data.
> 
> What's the scenario where we need these objects before the request completes? If none currently, this seems like unnecessary complexity and I would argue for waiting to create the objects until after the openai request completes.

I am still seeing this as a potential concern. min bar should for pilot launch should probably be that the UI does not break in case it loads the rubric UI while we are in this state.

also I am wondering if I am perhaps missing something, because I do not see `UNKNOWN` in https://github.com/code-dot-org/code-dot-org/blob/a705e0932b05465be4e1c3d6a7ec32cb55b9dd90/lib/cdo/shared_constants.rb#L601-L608

lastly, while my _personal_ (and possibly not fully-informed) preference is to wait to create these records until the aiproxy response is received, it should be a very quick fix on the UI to properly handle the new UNKNOWN case, so I'm also totally fine with that route and would be happy to push some commits to this branch to make that happen if that would be helpful (once I can see UNKNOWN in shared constants).

Reply:
oh wait, or is this code just orphaned as part of your latest refactor? maybe you can simply remove this block now that you are going through the RAI to find the LGAIs (apologies for the acronyms)

Reply:
It should be! I'm removing it and the UNKNOWN part. You will get back an empty array when you query a pending evaluation. There is no means of querying the rubric evaluation status.